File: 20-ValidParentheses/20-ValidParentheses.py
# Last updated: 8/20/2025, 5:44:17 PM
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        mapping = {')': '(', '}': '{', ']': '['}
        
        for char in s:
            if char in mapping.values():  # if it's an opening bracket
                stack.append(char)
            elif char in mapping:  # if it's a closing bracket
                if not stack or stack[-1] != mapping[char]:
                    return False
                stack.pop()
            else:
                return False
        
        return not stack

File: 20-ValidParentheses/20-ValidParentheses.c
// Last updated: 8/20/2025, 5:44:24 PM
bool isValid(char* s)
{
    int len=strlen(s);
    char st[len];
    int top =-1;
    for(int i=0;i<len;i++)
    {
        if(s[i]=='('||s[i]=='{'||s[i]=='['){
               st[++top]=s[i];
        }
        else
        {
             if(top==-1||(s[i]==')'&&st[top]!='(')||(s[i]==']'&&st[top]!='[')||(s[i]=='}'&&st[top]!='{'))
             return false;
             else
             top--;
        }
    }
    if(top==-1)
    return true;
    else
    return false;
}